python/pyspark/sql/session.py

```diff
@@ -161,8 +161,8 @@ def getOrCreate(self):
             with self._lock:
                 from pyspark.context import SparkContext
                 from pyspark.conf import SparkConf
-                session = SparkSession._instantiatedContext
-                if session is None:
+                session = SparkSession._instantiatedSession
+                if session is None or session._sc._jsc is None:
```

Comment:
Yeah, I had to change to just checking whether _jsc was defined in our branch for the same reason. I'd prefer to add isStopped to Java and Python eventually, but that doesn't need to block this commit.